File: swt/src/lib/cio/getarg.f
      INTEGER FUNCTION GETARG(ARGNUM,EOSBUF,BUFLEN)
      INTEGER ARGNUM,BUFLEN
      INTEGER EOSBUF(BUFLEN)
      INTEGER INFO(8),BUFFER(80),CODE,I
      INTEGER PTOC,EQUAL
      INTEGER AAAAA0(2)
      INTEGER AAAAB0(4)
      DATA AAAAA0/242,0/
      DATA AAAAB0/243,229,231,0/
      CALL RDTK$$(3,INFO,BUFFER,80,CODE)
      CALL RDTK$$(2,INFO,BUFFER,80,CODE)
      CALL PTOC(BUFFER,160,EOSBUF,80)
      CALL MAPSTR(EOSBUF,1)
      IF((EQUAL(EOSBUF,AAAAA0).EQ.1))GOTO 10001
      IF((EQUAL(EOSBUF,AAAAB0).EQ.1))GOTO 10001
      GOTO 10000
10001   CALL RDTK$$(2,INFO,BUFFER,80,CODE)
10000 I=0
      GOTO 10004
10002 I=I+(1)
10004 IF((I.GE.ARGNUM))GOTO 10003
      IF((INFO(1).EQ.6))GOTO 10003
        CALL RDTK$$(2,INFO,BUFFER,80,CODE)
      GOTO 10002
10003 IF((INFO(1).NE.6))GOTO 10005
        GETARG=-1
        RETURN
10005 GETARG=PTOC(BUFFER,160,EOSBUF,80)
      RETURN
      END
C ---- Long Name Map ----

File: swt/MFD/SRC/LIB/CIO/GETARG.R
# getarg --- an interlude to grab primos command args

   integer function getarg (arg_num, eos_buf, buf_len)
   integer arg_num, buf_len
   character eos_buf (buf_len)

   define (RT_NEXT,2)
   define (RT_RESET,3)
   define (RT_EOF,6)

   integer info (8), buffer (80), code, i
   integer ptoc, equal

# Reset the pointer to the command line first

   call rdtk$$ (RT_RESET, info, buffer, 80, code)

# Check for a user or system program calling us
# If the first arg is an "R" or a "SEG", ignore it.

   call rdtk$$ (RT_NEXT, info, buffer, 80, code)
   call ptoc (buffer, ' 'c, eos_buf, 80)
   call mapstr (eos_buf, LOWER)
   if (equal (eos_buf, "r"s) == YES || equal (eos_buf, "seg"s) == YES)
        call rdtk$$ (RT_NEXT, info, buffer, 80, code)

# Now scan until we get the argument that we wanted

   for (i = 0; i < arg_num && info (1) ~= RT_EOF; i += 1)
      call rdtk$$ (RT_NEXT, info, buffer, 80, code)

# Now process the correct argument

   if (info (1) == RT_EOF)
      return (EOF)

   return (ptoc (buffer, ' 'c, eos_buf, 80))
   end
